File: flutter_app/lib/api/safety_api.dart
import 'api_client.dart';

class SafetyApi {
  SafetyApi(this._client);

  final ApiClient _client;

  Future<void> blockUser(int blockedId) async {
    await _client.post('/block', body: {
      'blocked_id': blockedId,
    });
  }

  Future<void> reportUser({
    required int reportedId,
    required String reason,
    String? details,
  }) async {
    await _client.post('/report', body: {
      'reported_id': reportedId,
      'reason': reason,
      'details': details,
    });
  }
}

File: frontend/src/api/safetyApi.ts
import { ApiClient } from './apiClient';

export class SafetyApi {
  constructor(private readonly client: ApiClient) {}

  async blockUser(blockedId: number): Promise<void> {
    await this.client.post('/block', {
      blocked_id: blockedId,
    });
  }

  async reportUser(payload: {
    reportedId: number;
    reason: string;
    details?: string;
  }): Promise<void> {
    await this.client.post('/report', {
      reported_id: payload.reportedId,
      reason: payload.reason,
      details: payload.details,
    });
  }
}
